tests/functions/test_case_when.py

```diff
@@ -0,0 +1,28 @@
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from janitor.functions import case_when
+
+def test_update_where_query():
+    """Test that function works with pandas query-style string expression."""
```

Comment:
something else that comes to mind, is testing if numpy select preserves the dtype of the column - specifically pandas extension arrays. Say the column is a category dtype, will numpy select preserve it? I dont think so. what if it is an integer column with pandas nullable integer. will it be preserved? I dont think so. I think it will revert to numpy's own dtype. Thinking about this makes me wonder if np.select is really the best tool for the job. I know an [issue](https://github.com/pandas-dev/pandas/issues/39154) has been raised in Pandas regarding `case_when`; however, it is still an issue, with no real traction.